src/mpi/datatype/typerep/src/typerep_yaksa_pack.c

```diff
@@ -17,6 +62,21 @@ int MPIR_Typerep_pack(const void *inbuf, MPI_Aint incount, MPI_Datatype datatype
     int mpi_errno = MPI_SUCCESS;
     int rc;
 
+    if (incount == 0) {
+        *actual_pack_bytes = 0;
+        goto fn_exit;
+    }
+
+    MPIR_Assert(datatype != MPI_DATATYPE_NULL);
+
+    /* fast-path for contig CPU buffers */
+    if (inoffset == 0) {
+        bool fast = fastpath_memcpy(inbuf, outbuf, datatype, incount, max_pack_bytes,
```

Comment:
Why do we try fastpath only if `inoffset==0`?

Reply:
That was one of the approximations I had to make to try to put the code in a single function.  The reasoning was that, if in offset is not zero it's a large message anyway, so the optimization shouldn't matter much.

Reply:
Can we pass `inbuf + inoffset` to the subroutine instead of `inbuf`? I see `inbuf` is always used with the offset in the original code. Same for outbuf and outoffset

Reply:
No, the code is incorrect -- see my comment above.  Fixing it.  But, in order to fix it, I'll need to pass the copy direction (pack vs. unpack) to the function.  So, I might as well pass the offset in there and decide whether to apply that to the inbuf or outbuf inside the function.

Reply:
Pushed a new version.  I have done no local testing (not even warning testing).  Sorry.  Maybe wait for a few minutes for jenkins to do the warning testing?

Reply:
I assume that you are fixing the warning errors. I will wait for you updates.

Reply:
Pushed another version.  Tested locally this time.  :-)

Reply:
@minsii Can you take a look at this again?

Reply:
Yes, I was looking at it. It looks good to me.